components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationRequestCache.java

```diff
@@ -62,4 +68,29 @@ public static AuthenticationRequestCache getInstance() {
         }
         return instance;
     }
+
+    public void addToCache(CacheKey key,CacheEntry entry){
+        String keyValue = ((AuthenticationRequestCacheKey)key).getResultId();
```

Comment:
Changed the the method signature to use String as the key, PR : https://github.com/wso2/carbon-identity/pull/937

The casting code segment was there because the code was refactored to use String as cache key instead of objects. But for the initial step the underlying cache key didn't remove from the code, eventually all the cache keys should be removed.